crates/nu-cli/src/completions/completer.rs

```diff
@@ -349,143 +347,175 @@ impl NuCompleter {
+                                break;
+                            }
+                            Completion::List(list) => {
+                                let mut completer = StaticCompletion::new(list);
+                                // Prioritize argument completions over (sub)commands
+                                suggestions
+                                    .splice(0..0, self.process_completion(&mut completer, &ctx));
+                                // We don't want to fallback to file completion here
+                                return suggestions;
                             }
                         }
+                    } else if let Some(command_wide_completer) = signature.complete {
+                        let completion = match command_wide_completer {
+                            Some(decl_id) => CommandWideCompletion::command(
+                                working_set,
+                                decl_id,
+                                element_expression,
+                                strip,
+                            ),
+                            None => self
+                                .engine_state
+                                .get_config()
+                                .completions
+                                .external
+                                .completer
+                                .as_ref()
+                                .map(|closure| {
+                                    CommandWideCompletion::closure(
+                                        closure,
+                                        element_expression,
+                                        strip,
+                                    )
+                                }),
+                        };
 
-                        // normal arguments completion
-                        let (new_span, prefix) =
-                            strip_placeholder_if_any(working_set, &span, strip);
-                        let ctx = Context::new(working_set, new_span, prefix, offset);
-                        let flag_completion_helper = || {
-                            let mut flag_completions = FlagCompletion {
-                                decl_id: call.decl_id,
-                            };
-                            self.process_completion(&mut flag_completions, &ctx)
+                        if let Some(mut completion) = completion {
+                            let ctx = Context::new(working_set, span, b"", offset);
+                            let results = self.process_completion(&mut completion, &ctx);
+
+                            if !completion.need_fallback {
+                                // Prioritize external results over (sub)commands
+                                suggestions.splice(0..0, results);
+                                return suggestions;
+                            }
```

Comment:
I know that `need_fallback` goes hand in hand with empty results. But I would prefer moving the `splice` call out of the if block for better readability.